File: pset6/credit/credit.py
from cs50 import get_int

while True:
    number = get_int("Number: ")
    if number > 999999999 and number < 10000000000000000:
        break

s = 0
n1 = 0
n2 = 0
n21 = 0
mn = number
i = 0

while mn > 0:
    n1 = mn % 10
    s = s + n1
    mn = mn // 10
    i += 1
    if mn == 0:
        break
    n21 = mn % 10
    n2 = n21 * 2
    if n2 > 9:
        n2 = (n2 % 10) + (n2 // 10)
    s = s + n2
    mn = mn // 10
    i += 1
    if mn == 0:
        break

if i < 13 or i > 16:
    print("INVALID\n")
    exit()

if s % 10 != 0:
    print("INVALID\n")
    exit()

if i % 2 == 0:
    if n21 == 4:
        print("VISA\n")
        exit()
    elif n21 == 5:
        if n1 > 0 and n1 < 6:
            print("MASTERCARD\n")
            exit()
    else:
        print("INVALID\n")
        exit()
else:
    if n1 == 4:
        print("VISA\n")
        exit()
    elif n1 == 3:
        if n21 == 4 or n21 == 7:
            print("AMEX\n")
            exit()
    else:
        print("INVALID\n")
        exit()

File: pset1/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long number;

    do
    {
        number = get_long("Number: ");
    }
    while (number < 999999999999 && number > 10000000000000000);
    
    int sum = 0;
    int num_1 = 0;
    int num_2 = 0;
    int num_2_1 = 0;
    long mod_num = number;
    int i = 0;    

    while (mod_num > 0)
    {
        num_1 = mod_num % 10;
        sum = sum + num_1;
        mod_num = mod_num / 10;
        i++;
        if (mod_num == 0)
        {
            break;
        } 
        num_2_1 = mod_num % 10;
        num_2 = num_2_1 * 2;
        if (num_2 > 9)
        { 
            num_2 = (num_2 % 10) + (num_2 / 10);
        }
        sum = sum + num_2;
        mod_num = mod_num / 10;
        i++;
        if (mod_num == 0)
        {
            break;
        } 
    }
    if (i < 13 || i > 16)
    {
        printf("INVALID\n");
        return 0;
    }
    if (sum % 10 != 0)
    {
        printf("INVALID\n");
        return 0;
    }
    if (i % 2 == 0)
    {
        switch (num_2_1)
        {
            case 4:
                printf("VISA\n");
                break;

            case 5:
                if (num_1 > 0 && num_1 < 6)
                {
                    printf("MASTERCARD\n");
                    break;
                }
            default:
                printf("INVALID\n");
                break;
        }
    }
    else
    {
        switch (num_1)
        {
            case 4:
                printf("VISA\n");
                break;

            case 3:
                if (num_2_1 == 4 || num_2_1 == 7)
                {
                    printf("AMEX\n");
                    break;
                }
            default:
                printf("INVALID\n");
                break;
        }
    }
}
